pkg/cmd/prerunner.go

```diff
@@ -190,6 +190,10 @@ func (r *PreRun) Authenticated(command *AuthenticatedCLICommand) func(*cobra.Com
 		setContextErr := r.setAuthenticatedContext(command)
 		if setContextErr != nil {
 			if _, ok := setContextErr.(*errors.NotLoggedInError); ok {
+				if cmd.Use == "logout" {
```

Comment:
I tried some other scenarios and it looks like other errors can surface. If the login token expires, `logout` throws an `expired token` error, and if you try logging out again after that you get the `not logged in` error again (which is because although our current context exists, but we're no longer considered logged into it). These errors result from the fact that we're trying to have `logout` work exactly the same as before when the user is not logged in, despite it now being an `authenticatedCLICommand`.

I think we should probably just wait until v4 to add this since it doesn't look like all the possible breaking changes from errors can be avoided without putting several of these hacky exceptions for the `logout` command in the prerunner.

Reply:
> I think we should probably just wait until v4 to add this since it doesn't look like all the possible breaking changes from errors can be avoided without putting several of these hacky exceptions for the logout command in the prerunner.

Labeling as v4!

Reply:
Going to revert the last two commits since we want the breaking changes to be visible.